sled-agent/src/probe_manager.rs

```diff
@@ -118,43 +120,10 @@ impl ProbeManager {
         }
     }
 
-    /// Add a probe to the manager.
-    ///
-    /// This will ensure that a corresponding zone is created, asynchronously.
-    /// An error is returned if the probe already exists.
-    pub(crate) fn add_probe(&self, params: ProbeCreate) -> Result<(), Error> {
-        let id = params.id;
-        if self.expected_probes_tx.send_if_modified(|expected| {
-            match expected.entry(id) {
-                Entry::Occupied(_) => false,
-                Entry::Vacant(entry) => {
-                    entry.insert(ProbeState::from(params));
-                    true
-                }
-            }
-        }) {
-            Ok(())
-        } else {
-            Err(Error::ProbeAlreadyExists(id))
-        }
-    }
-
-    /// Remove a probe from the manager.
-    ///
-    /// This will ensure that the requested zone will be removed,
-    /// asynchronously. Note that this is infallible, and doesn't fail if the
-    /// probe was already gone.
-    pub(crate) fn remove_probe(&self, id: Uuid) {
-        self.expected_probes_tx
-            .send_if_modified(|expected| expected.remove(&id).is_some());
-    }
-
     /// Completely replace the set of managed probes.
-    pub(crate) fn set_probes(&self, probes: Vec<ProbeCreate>) {
-        let probes = probes
-            .into_iter()
-            .map(|probe| (probe.id, ProbeState::from(probe)))
-            .collect();
+    pub(crate) fn set_probes(&self, probes: IdHashMap<ProbeCreate>) {
+        let probes =
+            probes.into_iter().map(|probe| ProbeState::from(probe)).collect();
```

Comment:
I don't think we need these two lines at all now, right? `probes` is already the correct type.

Reply:
Not quite, the input is a `ProbeCreate` while the watch channel holds a bunch of `ProbeState`.

Reply:
Ahhh sorry, missed that. Carry on!